extensions-contrib/sqlserver-metadata-storage/src/main/java/io/druid/metadata/storage/sqlserver/SQLServerConnector.java

```diff
@@ -0,0 +1,183 @@
+package io.druid.metadata.storage.sqlserver;
+
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.dbcp2.BasicDataSource;
+import org.skife.jdbi.v2.Binding;
+import org.skife.jdbi.v2.ColonPrefixNamedParamStatementRewriter;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.StatementContext;
+import org.skife.jdbi.v2.tweak.HandleCallback;
+import org.skife.jdbi.v2.tweak.RewrittenStatement;
+import org.skife.jdbi.v2.tweak.StatementRewriter;
+import org.skife.jdbi.v2.util.StringMapper;
+
+import com.google.common.base.Supplier;
+import com.google.inject.Inject;
+import com.metamx.common.logger.Logger;
+
+import io.druid.metadata.MetadataStorageConnectorConfig;
+import io.druid.metadata.MetadataStorageTablesConfig;
+import io.druid.metadata.SQLMetadataConnector;
+
+@SuppressWarnings( "nls" )
+public class SQLServerConnector extends SQLMetadataConnector
+{
+  private static final Logger log = new Logger(SQLServerConnector.class);
+
+  private static final String PAYLOAD_TYPE = "VARBINARY(MAX)";
+  private static final String SERIAL_TYPE = "[int] IDENTITY (1, 1)";
+
+  public static final int DEFAULT_STREAMING_RESULT_SIZE = 100;
+
+  private final DBI dbi;
+
+
+  private final Set<String> TRANSIENT_SQL_CLASS_CODES = new HashSet<>(Arrays.asList(
+    "08", "53", "54", "57", "58",   // Resource Failures
+    "JW", "JZ", "S1",               // Transient Failures
+    "40"                            // Transaction Rollback
+  ));
+
+  @Inject
+  public SQLServerConnector(Supplier<MetadataStorageConnectorConfig> config, Supplier<MetadataStorageTablesConfig> dbTables)
+  {
+    super(config, dbTables);
+
+    final BasicDataSource datasource = getDatasource();
+    datasource.setDriverClassLoader(getClass().getClassLoader());
+    datasource.setDriverClassName("com.microsoft.sqlserver.jdbc.SQLServerDriver");
+
+    this.dbi = new DBI(datasource);
+
+    this.dbi.setStatementRewriter( new CustomStatementRewriter() );
+
+    log.info("Configured Sql Server as metadata storage");
+  }
+
+  public static class CustomStatementRewriter implements StatementRewriter
+  {
+      @Override
+    public RewrittenStatement rewrite(String sql, Binding params, StatementContext ctx)
+    {
+
+      String sqlUpdated = sql
+        .replaceAll( "(?i)BOOLEAN NOT NULL DEFAULT FALSE", "BIT NOT NULL DEFAULT (0)" )
+        .replaceAll( "(?i)BOOLEAN NOT NULL DEFAULT TRUE", "BIT NOT NULL DEFAULT (1)" )
+        .replaceAll( "(?i)BOOLEAN DEFAULT FALSE", "BIT NOT NULL DEFAULT (0)" )
+        .replaceAll( "(?i)BOOLEAN DEFAULT TRUE", "BIT NOT NULL DEFAULT (1)" )
+        .replaceAll( "(?i)BOOLEAN", "BIT" )
+        .replaceAll( "(?i)TRUE", "1" )
+        .replaceAll( "(?i)FALSE", "0" )
+        ;
+      return ( new ColonPrefixNamedParamStatementRewriter() ).rewrite( sqlUpdated, params, ctx );
```

Comment:
you probably don't need to create a new instance of `ColonPrefixNamedParamStatementRewriter` every single time? That way you could benefit from the cache it maintains.